### PR TITLE
feat(client-go): add domains endpoint

### PR DESCRIPTION
**Pull request by Joshua-Anderson**

### CODE REVIEW

File: client-go/controller/models/domains/domains.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+	domains := api.Domains{}
+	if err = json.Unmarshal([]byte(body), &domains); err != nil {
+		return []api.Domain{}, err
+	}
+
+	return domains.Domains, nil
+}
+
+// New adds domains to an app.
```
Comment (mboersma) [inline client-go/controller/models/domains/domains.go]: Only one though, so maybe the comment should be singular.
```
// New adds a domain to an app.
```

File: client-go/controller/models/domains/domains.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+	res := api.Domain{}
+	if err = json.Unmarshal([]byte(resBody), &res); err != nil {
+		return api.Domain{}, err
+	}
+
+	return res, nil
+}
+
+// Delete removes domains registered with an app.
```
Comment (mboersma) [inline client-go/controller/models/domains/domains.go]: Same as above, I think singular is more clear:
```
// Delete removes a domain registered with an app.
```

**Comment by Joshua-Anderson** [comment]
@mboersma Done ( I was already fixing it as you commented)!

File: client-go/parser/domains.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
+package parser
+
+import (
+	"fmt"
+
+	"github.com/deis/deis/client-go/cmd"
+	docopt "github.com/docopt/docopt-go"
+)
+
+// Domains routes domain commands to the specific function
```
Comment (mboersma) [inline client-go/parser/domains.go]: Ideally all comments should be full sentences, ending with a period, so:
```
// Domains routes a domain command to the specific function implementing it.
```

**Comment by mboersma** [comment]
Code LGTM, my comments about comments notwithstanding.

File: client-go/parser/domains.go (hunk 3 above)
Comment (Joshua-Anderson) [inline client-go/parser/domains.go]: Already fixed, refresh!

**Comment by technosophos** [comment]
Code LGTM

**Pull request merged**